src/ResourceManager/Compute.ManagedService/Commands.Compute.ManagedService/StorageServices/NewAzureVhdVMCommand.cs

```diff
@@ -0,0 +1,576 @@
+        [Parameter(ParameterSetName = DiskFileParameterSetNameStr, Mandatory = true, Position = 2, ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet("Windows", "Linux")]
+        public string OSType { get; set; }
+        
+        [Parameter(ParameterSetName = DiskLinkParameterSetNameStr, Mandatory = true, Position = 3, ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string[] DiskLink { get; set; }
+        
+        [Parameter(ParameterSetName = DiskFileParameterSetNameStr, Mandatory = true, Position = 3, ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string[] DiskFile { get; set; }
+
+        [Parameter(ParameterSetName = DiskLinkParameterSetNameStr, Mandatory = false, Position = 4, ValueFromPipelineByPropertyName = true)]
+        [Parameter(ParameterSetName = DiskFileParameterSetNameStr, Mandatory = false, Position = 4, ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string VMName { get; set; }
+
+        [Parameter(ParameterSetName = DiskLinkParameterSetNameStr, Mandatory = false, Position = 5, ValueFromPipelineByPropertyName = true)]
+        [Parameter(ParameterSetName = DiskFileParameterSetNameStr, Mandatory = false, Position = 5, ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string VMSize { get; set; }
+        
+        [Parameter(ParameterSetName = DiskLinkParameterSetNameStr, Mandatory = false, Position = 6, ValueFromPipelineByPropertyName = true)]
+        [Parameter(ParameterSetName = DiskFileParameterSetNameStr, Mandatory = false, Position = 6, ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public List<SecurityRule> SecurityRules { get; set; }
+
+        [Parameter(ParameterSetName = DiskLinkParameterSetNameStr, Mandatory = false, Position = 7, ValueFromPipelineByPropertyName = true)]
+        [Parameter(ParameterSetName = DiskFileParameterSetNameStr, Mandatory = false, Position = 7, ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string DefaultVNetAddressSpace { get; set; }
+
+        [Parameter(ParameterSetName = DiskLinkParameterSetNameStr, Mandatory = false, Position = 8, ValueFromPipelineByPropertyName = true)]
+        [Parameter(ParameterSetName = DiskFileParameterSetNameStr, Mandatory = false, Position = 8, ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string DefaultSubnetAddressSpace { get; set; }
+
+        [Parameter(ParameterSetName = DiskLinkParameterSetNameStr, Mandatory = false, Position = 9, ValueFromPipelineByPropertyName = true)]
+        [Parameter(ParameterSetName = DiskFileParameterSetNameStr, Mandatory = false, Position = 9, ValueFromPipelineByPropertyName = true)]
```

Comment:
remove position for optional parameters.

Reply:
What's the default behavior of having optional parameters without positions? Do they require ```-XXX $xxx``` usage, if the position indexes are removed?

Reply:
Without position, the parameter requires its parameter name, like -XXX.  We don't recommend a position parameter greater than 5, since it is difficult for users to memorize such many positions.    Also, for optional parameters, position may confuse users, because users must give all parameter values up to the certain position.  (So, if position 9 is optional parameter, then to use this parameter as position parameter, a user give all parameters up to position 9, even though some of them are optional.)  So we don't recommend a position for optional parameters.

Reply:
ok